core/src/main/java/org/apache/brooklyn/core/typereg/RegisteredTypePredicates.java

```diff
@@ -109,36 +109,46 @@ public boolean apply(@Nullable RegisteredType item) {
         }
     }
 
-    public static <T> Predicate<RegisteredType> javaType(final Predicate<Class<T>> filter) {
-        return new JavaTypeMatches(filter);
+    public static <T> Predicate<RegisteredType> anySuperType(final Predicate<Class<T>> filter) {
```

Comment:
i don't think nested generic wildcard works like that -- `Predicate<Class<?>>` does not match any `Predicate<Class<T>>`.